### ISSUE TITLE
False-positive "No input file found for <file-with-issues>" for project with multiple subprojects

### ISSUE THREAD

**Issue by dominikmontada** [SonarSource/SonarJS#1124]
**I want to report a bug.**

**SonarJS version:** 5.0 (build 6962)
**SonarQube version:** 7.4.0.18908<br>

**Info** 
I have an ASP.NET Core with Vue.js project that I am analyzing with Sonarqube. C# sources are built with `dotnet build <solution>` and Vue.js parts are built with yarn and webpack.

I want to import ESLint issues, however it says `No input file found for <file with issues>` after importing the ESLint report successfully. Normal SonarJS analyses work as expected, only ESLint imports are failing.

ESLint report is created with `eslint -f json -o report.json .`

**Logs**
```
INFO: Sensor Import of ESLint issues [javascript]
INFO: Importing /appname/src/report.json
WARNING: WARN: No input file found for /appname/src/main.js. No ESLint issues will be imported on this file.
INFO: Sensor Import of ESLint issues [javascript] (done) | time=3ms
INFO: Sensor C# Properties [csharp]
WARNING: WARN: No input file found for /appname/src/util/endpoints.js. No ESLint issues will be imported on this file.
```

**Expected behavior**

ESLint issues get imported to Sonarqube after importing report file.

**Comment by dominikmontada** [comment]
**Additional information**
This is the (simplified) script that I am running to build the project and run sonar-scanner

```bash
dotnet sonarscanner begin /k:"$SONARQUBE_PROJ_KEY:$BRANCH" /n:"$SONARQUBE_PROJ_NAME" /d:sonar.login=$SONARQUBE_TOKEN /d:sonar.host.url="$SONARQUBE_HOST" /d:sonar.eslint.reportPaths=appname/report.json

dotnet restore Solution.sln
dotnet build --no-restore Solution.sln
dotnet test --no-restore --no-build /p:CollectCoverage=true /p:CoverletOutputFormat=opencover /p:CoverletOutput=./bin/Coverage/ Solution.sln

cd appname
yarn install
yarn run lint
cd ..

dotnet sonarscanner end /d:sonar.login=$SONARQUBE_TOKEN
```

**Comment by vilchik-elena** [comment]
As far as I understood, paths in your eslint report are relative. Are they relative to SQ root project directory?

**Comment by dominikmontada** [comment]
@vilchik-elena Sorry, that was unclear from my part as I tried to hide the true folder names. The paths inside the eslint are absolute, not relative. I will edit my previous post to fix that.

**Comment by vilchik-elena** [comment]
Do you see file `/appname/src/main.js` in SQ UI? It should be part of your project so that we could add eslint issues to it.

**Comment by dominikmontada** [comment]
Yes the file is present in SQ UI and already shows the normal SonarJS feedback. Only ESLint import mysteriously reports the file as not existing
________________________________

**Comment by dominikmontada** [comment]
I believe I found the issue. After going through the log carefully, I noticed that ESLint issues are in fact imported successfully! The problem is that I have a project with multiple subprojects and the ESLint report is only relevant for one of these subprojects.

What happens is that the report is picked up by SonarJS for all subprojects. For the relevant subproject everything is fine and issues are imported correctly. For all other, irrelevant subprojects however, the files listed in the report obviously do not exist (or rather, they do exist but in another subproject), so SonarJS gives an error that these files cannot be found in the current project. So, it's just a false-positive.
